posts/2021-07-08-logging-format-21008-beta.adoc

```diff
@@ -0,0 +1,274 @@
+
+[source, java]
+----
+WLP_LOGGING_MESSAGE_FORMAT=TBASIC
+WLP_LOGGING_CONSOLE_FORMAT=TBASIC
+----
+
+
+TBASIC Logs Example:
+
+[source, java]
+----
+[24/03/21 15:04:10:331 EDT] 00000001 FrameworkMana A   CWWKE0001I: The server defaultServer has been launched.
+[24/03/21 15:04:11:338 EDT] 00000001 FrameworkMana I   CWWKE0002I: The kernel started after 1.177 seconds
+[24/03/21 15:04:11:465 EDT] 0000003e FeatureManage I   CWWKF0007I: Feature update started.
+[24/03/21 15:04:11:635 EDT] 00000033 DropinMonitor A   CWWKZ0058I: Monitoring dropins for applications.
+----
+
+
+[#staleConnectionID]
+=== Customizing Stale Connection Identification
+
+Open Liberty maintains a pool of JDBC connections to improve performance. It is necessary for Open Liberty to be able to identify when connections have become stale and are no longer usable so that such connections can be removed from the pool. Open Liberty leverages multiple standards made available by the JDBC and SQL specifications, as well as relying on some built-in knowledge of vendor-specific behavior for some JDBC drivers in order to achieve this.
+
+Not all JDBC drivers completely follow the JDBC/SQL specifications in identifying stale connections. If you are using such a JDBC driver, it is now possible for you to provide additional configuration for a data source that helps identify the vendor-specific SQL states and error codes that are raised by the JDBC driver, enabling Liberty to better maintain the connection pool.
+
+
+Configure one or more `<identifyException>` subelements under `<dataSource>` to provide the `SQLException` identification detail.
+
+[source,xml]
+----
+<featureManager>
+  <feature>jdbc-4.2</feature>
+  <feature>jndi-1.0</feature>
+  ... other features
+</featureManager>
+
+<dataSource id="DefaultDataSource" jndiName="jdbc/myDataSource">
+    <jdbcDriver libraryRef="myJDBCLib"/>
+    <properties databaseName="TESTDB" serverName="localhost" portNumber="1234"/>
+    <!-- identify the following as stale connections, -->
+    <identifyException sqlState="08000" as="StaleConnection"/>
+    <identifyException errorCode="2468" as="StaleConnection"/>
+    <!-- remove built-in identification of SQL state S1000 -->
+    <identifyException sqlState="S1000" as="None"/>
+</dataSource>
+
+<library id="myJDBCLib">
+    <file name="C:/drivers/some-jdbc-driver.jar"/>
+</library>
+----
+
+
+[#JEE9.1]
+=== Jakarta EE 9.1 convenience features
+
+Jakarta EE 9.1 convenience features are included in this beta release in the "all beta" zip file, but not yet in the Jakarta EE 9 beta zip file. The 9.1 features are duplicates of the Jakarta EE 9.0 convenience features. In future, the 9.0 convenience features will be removed and only the 9.1 ones will remain. Users of the Jakarta EE 9 convenience features should look to transition to use 9.1 in preparation of 9.0 being removed.
+
+To enable Jakarta EE 9.1 features you would add them to your server.xml or client.xml feature list such as:
```

Comment:
I would like the work `convenience` features added to between `9.1` and `features`.  We don't want the reader to think the only way to enable Jakarta EE 9.1 features is to use the convenience feature configuration.  They can do each individual feature which is of course preferred to not enable more features than needed to run your application.  I would also add the word `the` before `Jakarta`.

Reply:
Done

Reply:
Did you purposely not do the `the` change or was that an oversight?  Also should there be a comma after `features`?

Reply:
Sorry missed the second part of the comment.